src/omv/sensors/lepton.c

```diff
@@ -245,6 +245,7 @@ static int ioctl(sensor_t *sensor, int request, va_list ap) {
             float *arg_max_temp = va_arg(ap, float *);
             float min_temp_range = (lepton.high_temp_mode) ? LEPTON_MIN_TEMP_HIGH : LEPTON_MIN_TEMP_NORM;
             float max_temp_range = (lepton.high_temp_mode) ? LEPTON_MAX_TEMP_HIGH : LEPTON_MAX_TEMP_NORM;
+            // Don't use clamp here, the order of comparison is important.
             lepton.min_temp = IM_MAX(IM_MIN(*arg_min_temp, *arg_max_temp), min_temp_range);
             lepton.max_temp = IM_MIN(IM_MAX(*arg_max_temp, *arg_min_temp), max_temp_range);
```

Comment:
This `IM_MIN(*arg_min_temp, *arg_max_temp)` is finding the min of the two values. Since they are passed by the user they could be reversed. If you used clamp the order of operations works for

`lepton.min_temp = IM_MAX(IM_MIN(*arg_min_temp, *arg_max_temp), min_temp_range);`

but breaks for computing the second line since IM_MAX needs to be done first as you want to compare the final value choose against the max_temp_range.

Reply:
> Since they are passed by the user they could be reversed

This shouldn't be supported, we should assert that `min < max`, not support reversed min/max which is a bug.